x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -254,14 +327,20 @@ export function XYChart({ data, args, formatFactory, timeZone, chartTheme }: XYC
           const columnToLabelMap: Record<string, string> = columnToLabel
             ? JSON.parse(columnToLabel)
             : {};
+
           const table = data.tables[layerId];
+          const rows = table.rows.filter(
+            row =>
+              !(splitAccessor && !row[splitAccessor] && accessors.every(accessor => !row[accessor]))
```

Comment:
Could we please leave a comment in code, what that line does exactly, and why we need to filter down the rows here.